Initial implementation

somebody1234 opened the pull request:
To set up, run `pnpm install` (requires `pnpm` to be installed).
To run locally, run `pnpm run build; pnpm run start`.
To test development mode, run `pnpm run dev`.

A `config.json` is required at the root of the repository; here the configuration I have been using for testing:
```json
{
  "botName": "Enso Support",
  "websocketPort": 8082,
  "storageLocation": "./db.sqlite",
  "discordApplicationId": "1093457425962639422",
  "discordPublicKey": "79bd2c3d3e29a5d8409973207299c7a312274f2c26807e4293ab062a2b5a3362",
  "discordToken": "MTA5MzQ1NzQyNTk 2MjYzOTQyMg.GVHusy.i708SBGe-BuVWmJ73Y4274lwdJCxW9fnwvaH-c",
  "discordOauthClientId": "1093457425962639422",
  "discordOauthClientSecret": "hKOjAEWuPUOhWolfxJSYHzqo1dcEtBo3",
  "discordServerId": "1105875541694885928",
  "discordChannelId": "1105875542378545305",
  "discordStaffRoleId": "1123532192400818236"
}
```
(remove the space in `discordToken`)
:warning: These credentials SHOULD NOT be used for the actual bot, as they are publicly known.
Here is an invite to the testing server: https://discord.gg/cruJQaRxMn - note that you need to register with the bot (assign yourself staff role, and then DM your full name + an attached image to the bot)

somebody1234 committed:
Initial code

somebody1234 committed:
Remove chat command handling; implement registering staff with bot

somebody1234 committed:
Extract port out to config

somebody1234 committed:
Fix `npm run build` + `npm run start`; fix eslint config

somebody1234 commented:
@wdanilo I've notices that this repository is public - just want to confirm that this is intentional?

somebody1234 committed:
Scrollback

somebody1234 committed:
Reactions

wdanilo commented:
> @wdanilo I've notices that this repository is public - just want to confirm that this is intentional?

Thanks for double checking! ❤️ 
I believe we want this to be public. In fact, I want 100% of code to be public, but until now, we keep the cloud one closed. I don't see a reason to close this one. Are there any security concerns regarding this decision? If there are any keys, they should be stored as GitHub secrets to be non accessible outside of the build system.

wdanilo commented on database.ts:
```diff
@@ -0,0 +1,329 @@
+// =============
+
+/** All possible emojis that can be used as a reaction on a chat message. */
+export type ReactionSymbol = '❤️' | '🎉' | '👀' | '👍' | '👎' | '😀' | '🙁'
```
This is duplicated in another file. Please double check the code for duplications.

somebody1234 replied:
this is intentional - not sure it makes too much sense to import `database.ts` from `chat.ts` or vice versa...
(f course there's no real problem, especially since it gets bundled anyway, and there should be no risk of a circular dependency since they don't *actually* depend on each other)

i could extract this (`ReactionSymbol`) to its own file, maybe?

wdanilo replied:
Yeah, if these two modules does not make sense to import each other, then refactoring it to another file is probably the best solution! :)

wdanilo reviewed (approved):
Really nice code. Side note – I'm always terrified seeing aslant config. It's sooooo complex! I'm sad it can't be simpler :(

somebody1234 committed:
Add missing `devDependency`

somebody1234 committed:
Rename `build` script

somebody1234 committed:
Address review

somebody1234 commented:
force-pushing to add an export of `ReactionSymbol` to `chat.ts` (minor change)

somebody1234 committed:
Remove `CONFIG` import from `chat.ts`

somebody1234 committed:
Reactor database schema out to a separate file

This is required to avoid unnecessary dependencies on modules like `better-sqlite3` on `chat.ts`
